recipes/gatk4/meta.yaml

```diff
@@ -12,67 +12,99 @@ source:
 outputs:
   - name: gatk4
     script: build_main.sh
     requirements:
+      build:
+        - conda-forge::gcc_impl_linux-64=7.5.0=habd7529_20
+        - conda-forge::gcc_linux-64=7.5.0=h47867f9_33
+        - conda-forge::gfortran_impl_linux-64=7.5.0=h56cb351_20
+        - conda-forge::gfortran_linux-64=7.5.0=h78c8a43_33
+        - conda-forge::gxx_impl_linux-64=7.5.0=hd0bb8aa_20
+        - conda-forge::gxx_linux-64=7.5.0=h555fc39_33
+        - conda-forge::glib=2.66.3=h58526e2_0
+        - conda-forge::python=3.6.10=h8356626_1011_cpython
+        - unzip
+        - wget 
+      host:
+        - absl-py=0.15.0
+        - conda-forge::pip=21.3.1
+        - conda-forge::mkl=2019.5
+        - conda-forge::mkl-service=2.3.0
+        - conda-forge::joblib=1.1.1
+        - conda-forge::numpy=1.17.5
+        - conda-forge::theano=1.0.4=py36h831f99a_1002
+        - defaults::tensorflow=1.15.0=mkl_py36*
+        - conda-forge::scipy=1.0.0
+        - conda-forge::pymc3=3.1
+        - conda-forge::h5py=2.10.0
+        - defaults::_tflow_select=2.3.0=mkl
+        - conda-forge::keras=2.2.4
+        - defaults::intel-openmp=2019.4
+        - conda-forge::scikit-learn=0.23.1
+        - conda-forge::matplotlib=3.2.1
+        - conda-forge::pandas=1.0.3
+        - conda-forge::typing_extensions=4.1.1
+        - conda-forge::dill=0.3.4
+        - conda-forge::gast=0.2.2=py_0
+        - conda-forge::biopython=1.76=py36h516909a_0
       run:
-        - openjdk >=17,<18
-        - python
-    test:
-      commands:
-        - gatk -h
-        - gatk --list
-        - gatk HaplotypeCaller --help
-        - gatk DetermineGermlineContigPloidy --help
-  - name: gatk4-spark
-    script: build_spark.sh
-    build:
-      noarch: generic
-    requirements:
-      run:
-        - openjdk >=17,<18
-        - python
-        - {{ pin_subpackage("gatk4", exact=True) }}
-    test:
-      commands:
-        - gatk -h
-        - gatk --list
-        - gatk MarkDuplicatesSpark --help
-        - gatk DetermineGermlineContigPloidy --help
+        - absl-py=0.15.0
+        - conda-forge::python =3.6.10
+        - conda-forge::pip=21.3.1
+        - conda-forge::mkl=2019.5
+        - conda-forge::mkl-service=2.3.0
+        - conda-forge::joblib=1.1.1
+        - conda-forge::numpy=1.17.5
+        - conda-forge::theano=1.0.4=py36h831f99a_1002
+        - defaults::tensorflow=1.15.0=mkl_py36*
+        - defaults::_tflow_select=2.3.0=mkl
+        - conda-forge::scipy=1.0.0
+        - conda-forge::pymc3=3.1
+        - conda-forge::h5py=2.10.0
+        - conda-forge::keras=2.2.4
+        - defaults::intel-openmp=2019.4
+        - conda-forge::scikit-learn=0.23.1
+        - conda-forge::matplotlib=3.2.1
+        - conda-forge::pandas=1.0.3
+        - conda-forge::typing_extensions=4.1.1
+        - conda-forge::dill=0.3.4
+        - r-base=3.6.2
```

Comment:
I can only recommend to relax your pinnings, this will cause problems down the road and will prevent conda users to install other tools in the same environment.

Reply:
GATK requires specific version of certain packages installed in the environment as you can see in their environment file here https://github.com/broadinstitute/gatk/blob/master/scripts/gatkcondaenv.yml.template. Since other versions aren't technically supported, I am worried about the results not being reproducible between different environments.